### DISCUSSION TITLE
search in Shaarli via web browser bar ( or something like searX)

### DISCUSSION THREAD

**Discussion by GannonTdW** [shaarli/Shaarli#2057] [Ideas]
Hello,

I've been using shaarli for quite a few years now, and I realize that I rarely have the reflex to search for a tag in shaarli before looking for it on a search engine.

Is there any way of getting my web browser's address bar to search my shaarli? 
In the same way as it searches my favorites and my search engine.

Or can I set up a search engine, such as [searX](https://searx.github.io/searx/blog/index.html), to do this?

Thanks for your help.

**Comment by virtadpt** [comment]
It should be possible to use Shaarli's API in Searx to carry out searches.  I haven't done it yet but I do have a bot which does; the principle should be the same.

https://github.com/virtadpt/exocortex-halo/blob/master/shaarli_bot/search.py

Mostly, the trick is getting Searx's templates to do what you want.

**Comment by ArthurHoaro** [comment]
Hi, Shaarli has an API that's compatible with browser's search and can be added as a search engine. It should work out of the box, example on Firefox:

![image](https://github.com/shaarli/Shaarli/assets/1962678/8af9d15c-d84f-4ada-9bea-28c81fe6d0ca)

I'm not sure that it's documented? cc @nodiscc

**Comment by nodiscc** [comment]
> I'm not sure that it's documented?

It's not. I will make a note to document it in https://shaarli.readthedocs.io/en/master/Usage.html#search. (edit: https://github.com/shaarli/Shaarli/pull/2096)

In short:
- Access your Shaarli instance in a Web browser
- In your browser's search bar, click the `Add search engine "YOUR SHAARLI NAME"` button that should display alongside other search engines
- You can now search your Shaarli bookmarks directly from your browser's search bar (simply type a few search terms, and select this search engine).

Note: on Firefox you can also set a "shortcut" key for each engine from the settings page - for example setting the `s` shortcut for Shaarli, and typing `s something` will search for `something` on your Shaarli instance.

**Comment by GannonTdW** [comment]
Thanks for your answers.
I had not seen that it was already possible.
So I can search in my Shaarli directly via the search bar of my web browser.

However  after my tests, search look are only done on my Shaarli if we select Shaarli.
Is it possible that searches (or rather suggestions ) are done on at the same time on my search engine, my Firefox favorites and my Shaarli ?
It's work already with favorites and search engine.

**Reply by nodiscc** [reply]
You mean search suggestions/search-as-you-type results?

Currently not possible.

It would need to be implemented from scratch https://devdoc.net/web/developer.mozilla.org/en-US/docs/Supporting_search_suggestions_in_search_plugins.html

**Reply by GannonTdW** [reply]
:(
OK, thanks.

Thanks for the link, I'll look into it.
